internal/util/validation/validation.go

```diff
@@ -107,6 +108,52 @@ func ValidateBearerToken(token *string, path string) []error {
 	return asErrors(errs)
 }
 
+// usages for a CSR should include 'client auth'
+// TODO - do we want to add more here? or just not let the user populate this?
+func ValidateUsages(u *[]string) []error {
+	errs := field.ErrorList{}
+
+	for _, a := range *u {
+		if a == "client auth" {
+			return nil
+		}
+	}
+	errs = append(errs, field.Invalid(fieldPathFor("spec.usages"), u, "must contain 'client auth'"))
+	return asErrors(errs)
+}
+
+// signer name from https://github.com/flightctl/flightctl/blob/75cb5bfada1f1dbd995797edb5d8a67ac251b861/cmd/flightctl-api/main.go#L26
```

Comment:
How does the user know to put "ca" in that field?  Will it ever be something else?  If not, then why have the user provide it at all?

Reply:
IMHO the user should not provide a signer. It should be chosen by flightctl. 
Again - config and/or configurable policy.
We should check the CSR for some fields though - it should have CA:false in its usage, otherwise anyone with a signed CSR can generate certs which may authenticate correctly vs flightctl as they have a valid ca path.

Reply:
Agree, it doesn't make sense to have the user provide it currently since we only have the one signer. I think the long term vision was the "signer" field is how the user would request a certain type of certificate - enrollment cert is signed by the enrollment signer, vs. some other type of cert is some other signer.

I've added the validation as a placeholder, but we can make sure the signer field is filled by FC for now so the user doesn't get an error for no reason.

Reply:
That makes sense. Though in that case we will need to apply filtering/sanity checks to this field.
As far as requesting - that is exactly how it is requested in the order interface in certmanager. You have to specify a signer there.